server/src/main/java/io/crate/types/DataType.java

```diff
@@ -117,7 +124,7 @@ public T valueForInsert(Object value) {
         return (T) value;
     }
 
-    public abstract T value(Object value) throws IllegalArgumentException, ClassCastException;
+    public abstract T sanitizeValue(Object value);
```

Comment:
I think we should add some docs to clarify the difference between `implicitCast` and `sanitizeValue`